### PR TITLE
Feature/uxmcloud 1314 => Ip Load Balancer in cloud manager

### PR DESCRIPTION
**Pull request by Mboulianne**
## Ip Load Balancer

Add IPLB management in Cloud Manager.

### CODE REVIEW

File: client/app/iplb/home/iplb-home.controller.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        this.subscription.load();
+
+        this.iplbStatus.load();
+        this.frontendsStatus.load();
+        this.serverFarmsStatus.load();
+
+        this.initActions();
+        this.initGraph();
+
+        this.IPLBActions = [[{
```
Comment (Mboulianne) [inline client/app/iplb/home/iplb-home.controller.js]: Faudrait s'entendre sur une structure pour les actions.  Y'a celle-ci et celle plus bas.  Une utilisée par iplb-status-card et une utilisée par cui-tile (et ses enfants).

Sinon pourquoi c'est un component iplb vs cui?  Pcq ça sera propre à IPLB?

Comment (neolitec) [inline client/app/iplb/home/iplb-home.controller.js]: En effet pour iplb-status-card, je pense que c'est pas mal du spécifique pour IPLB. A voir avec nos amis CX mais je ne voulais pas polluer notre ui-components déjà bien rempli...

File: client/app/iplb/home/iplb-home.service.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+                        ],
+                        status: workingServers === totalServers ? "success" : "error"
+                    }
+                };
+            })
+            .catch(this.ServiceHelper.errorHandler("iplb_information_loading_error"));
+    }
+
+    getStatus () {
+        //  TODO: Do something
```
Comment (kumako) [inline client/app/iplb/home/iplb-home.service.js]: ??

File: client/app/iplb/frontends/iplb-frontends-edit.controller.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+    createFrontend () {
+        if (this.form.$invalid) {
+            return this.$q.reject();
+        }
+        this.saving = true;
+        this.CloudMessage.flushMessages();
+        return this.IpLoadBalancerFrontendsService.createFrontend(this.type, this.$stateParams.serviceName, this.getCleanFrontend())
+            .then(() => {
+                this.$state.go("network.iplb.detail.frontends");
```
Comment (whispyy) [inline client/app/iplb/frontends/iplb-frontends-edit.controller.js]: Function called so, you shouldn't need embrace.

File: client/app/iplb/iplb-metrics.service.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+        queries.forEach(query => {
+            _.assign(query, options);
+        });
+
+        return this.getToken().then(token => this.$http({
+            method: "POST",
+            url: [this.url, this.path].join("/"),
+            headers: {
+                // eslint-disable-next-line
```
Comment (kumako) [inline client/app/iplb/iplb-metrics.service.js]: cheater! :D

File: client/app/iplb/iplb-nat-ip.service.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+class IpLoadBalancerNatIpService {
+    constructor ($translate, IpLoadBalancing, ServiceHelper) {
+        this.$translate = $translate;
+        this.IpLoadBalancing = IpLoadBalancing;
+        this.ServiceHelper = ServiceHelper;
+    }
+
+    getNatIps (serviceName) {
+        return this.IpLoadBalancing.Lexi().natIp({ serviceName }).$promise
+            .then(response => response)
```
Comment (kumako) [inline client/app/iplb/iplb-nat-ip.service.js]: what's the utility of `then` here?

File: client/app/iplb/modal/nat-ip/iplb-nat-ip-detail.html [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<cui-modal>
+    <cui-modal-header data-on-dismiss="IpLoadBalancerNatIpDetailCtrl.dismiss()"></cui-modal-header>
+    <cui-modal-body title="'iplb_modal_nat_ip_title' | translate">
+        <cui-loader data-ng-if="IpLoadBalancerNatIpDetailCtrl.isModalLoading()"></cui-loader>
+        <div data-ng-if="!IpLoadBalancerNatIpDetailCtrl.isModalLoading()">
+            <cui-modal-text data-text="iplb_modal_nat_ip_description | translate"></cui-modal-text>
```
Comment (kumako) [inline client/app/iplb/modal/nat-ip/iplb-nat-ip-detail.html]: missing quotes for `iplb_modal_nat_ip_description`

File: client/app/iplb/modal/offer/iplb-offer-change.controller.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+            loaderFunction: () => this.IpLoadBalancerHomeService.getConfiguration(this.serviceName)
+        });
+    }
+
+    $onInit () {
+        this.configuration.load();
+    }
+
+    confirm () {
+        //  TODO : Do something here.
```
Comment (kumako) [inline client/app/iplb/modal/offer/iplb-offer-change.controller.js]: ??

File: client/app/iplb/modal/offer/iplb-offer-change.html [hunk 8]
```diff
@@ -0,0 +1,18 @@
+<cui-modal>
+    <form id="IplbOfferChangeForm">  
+        <cui-modal-header data-on-dismiss="IpLoadBalancerOfferChangeCtrl.cancel()"></cui-modal-header>
+        <cui-modal-body title="'iplb_modal_offer_change_title' | translate">
+            <cui-loader data-ng-if="IpLoadBalancerOfferChangeCtrl.isModalLoading()"></cui-loader>
+            <div data-ng-if="!IpLoadBalancerOfferChangeCtrl.isModalLoading()">
+                Old offer => {{ IpLoadBalancerOfferChangeCtrl.configuration.data.offer }}
+                <cui-modal-text data-text="'EDIT SOME OFFER HERE'"></cui-modal-text>
```
Comment (kumako) [inline client/app/iplb/modal/offer/iplb-offer-change.html]: wtf?

File: client/app/iplb/frontends/iplb-frontends-edit.controller.js [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+
+    updateFrontend () {
+        if (this.form.$invalid) {
+            return this.$q.reject();
+        }
+        this.saving = true;
+        this.CloudMessage.flushMessages();
+        return this.IpLoadBalancerFrontendsService.updateFrontend(this.type, this.$stateParams.serviceName, this.frontend.frontendId, this.getCleanFrontend())
+            .then(() => {
+                this.$state.go("network.iplb.detail.frontends");
```
Comment (whispyy) [inline client/app/iplb/frontends/iplb-frontends-edit.controller.js]: Same as previous comment.

File: client/app/iplb/home/iplb-home.controller.js (hunk 1 above)
Comment (Mboulianne) [inline client/app/iplb/home/iplb-home.controller.js]: Ça me va.  Dans le pire des cas on change le component de nom si on veut l'utiliser plus en large :).

File: client/app/iplb/home/iplb-home.controller.js [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+            text: this.$translate.instant("iplb_status_details"),
+            run: () => {
+                this.$state.go("network.iplb.detail.frontends");
+            }
+        }]];
+
+        this.farmsActions = [[{
+            text: this.$translate.instant("iplb_status_details"),
+            run: () => {
+                this.$state.go("network.iplb.detail.server-farm");
```
Comment (whispyy) [inline client/app/iplb/home/iplb-home.controller.js]: May be no need embrace here too.

File: client/app/iplb/server/iplb-server-edit.controller.js [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+        this.saving = true;
+        this.CloudMessage.flushMessages();
+        return this.IpLoadBalancerServerService.create(
+            this.farmType,
+            this.$stateParams.serviceName,
+            this.$stateParams.farmId,
+            this.getCleanServer()
+        )
+            .then(() => {
+                this.$state.go("network.iplb.detail.server-farm");
```
Comment (whispyy) [inline client/app/iplb/server/iplb-server-edit.controller.js]: Same as previous.

File: client/app/iplb/server/iplb-server-edit.controller.js [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+        this.CloudMessage.flushMessages();
+        return this.IpLoadBalancerServerService.update(
+            this.farmType,
+            this.$stateParams.serviceName,
+            this.$stateParams.farmId,
+            this.server.serverId,
+            this.getCleanServer()
+        )
+            .then(() => {
+                this.$state.go("network.iplb.detail.server-farm");
```
Comment (whispyy) [inline client/app/iplb/server/iplb-server-edit.controller.js]: Same.

File: client/app/iplb/serverFarm/iplb-server-farm-edit.controller.js [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+                type: ""
+            }
+        };
+        this.saving = false;
+        this.protocol = "http";
+        this.type = "http";
+        this.protocols = this.IpLoadBalancerConstant.protocols;
+        this.balances = this.IpLoadBalancerConstant.balances;
+        this.stickinesses = this.IpLoadBalancerConstant.stickinesses;
+        this.probeTypes = this.IpLoadBalancerConstant.probeTypes;
```
Comment (whispyy) [inline client/app/iplb/serverFarm/iplb-server-farm-edit.controller.js]: May be put it into the constructor ?

File: client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js [hunk 14]
```diff
@@ -1275,6 +1275,12 @@ angular.module("managerApp")
         if (diskType) {
             if (self.vmInEdition.status === "ACTIVE") {
                 var augmentedFlavor = addDetailsToFlavor(self.originalVm.flavor);
+
+                //Changing an existing instance's disktype from SSD to CEPH should never be possible.
```
Comment (Netnix) [inline client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js]: Why is it in this PR?

File: client/app/iplb/configuration/iplb-configuration.controller.js [hunk 15]
```diff
@@ -0,0 +1,66 @@
+class IpLoadBalancerConfigurationCtrl {
+    constructor ($stateParams, ControllerHelper, IpLoadBalancerConfigurationService,
+                 IpLoadBalancerZoneService) {
+        this.$stateParams = $stateParams;
+        this.ControllerHelper = ControllerHelper;
+        this.IpLoadBalancerConfigurationService = IpLoadBalancerConfigurationService;
+        this.IpLoadBalancerZoneService = IpLoadBalancerZoneService;
+
+        this.initLoaders();
```
Comment (Netnix) [inline client/app/iplb/configuration/iplb-configuration.controller.js]: The call is made during $onInit.

File: client/app/iplb/frontends/iplb-frontends-edit.html [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
+                data-value="protocol"
+                data-ng-disabled="ctrl.isProtocolDisabled(protocol)"></cui-radio-bar-item>
+          </cui-radio-bar>
+          <label for="protocol" class="oui-label" data-translate="iplb_frontend_add_protocol"></label>
+        </div>
+
+        <div class="oui-input-formfield">
+          <input type="number"
+            min="1"
+            max="65535"
```
Comment (Netnix) [inline client/app/iplb/frontends/iplb-frontends-edit.html]: From a const maybe?

File: client/app/iplb/frontends/iplb-frontends-edit.html [hunk 17]
```diff
@@ -0,0 +1,150 @@
...
+              <label for="http_redirection" class="oui-label" data-translate="iplb_frontend_add_http_redirection"></label>
+            </div>
+
+            <div class="oui-input-formfield" data-ng-if="ctrl.type == 'http'">
+              <input type="text" id="http_header" name="http_header" class="oui-input"
+                 data-ng-model="ctrl.frontend.httpHeader" />
+              <label for="http_header" class="oui-label" data-translate="iplb_frontend_add_http_header"></label>
+            </div>
+
+            <div class="oui-checkbox" data-ng-if="ctrl.type == 'http' && ctrl.frontend.ssl">
```
Comment (Netnix) [inline client/app/iplb/frontends/iplb-frontends-edit.html]: These is a component oui-checkbox in oui-angular that you might prefer from this one.

File: client/app/iplb/modal/failover-ip/iplb-failover-ip-detail.html [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+    <cui-modal-body title="'iplb_modal_failover_ip_title' | translate">
+        <cui-loader data-ng-if="IpLoadBalancerFailoverIpDetailCtrl.isModalLoading()"></cui-loader>
+
+        <cui-clipboard-list data-ng-if="!IpLoadBalancerFailoverIpDetailCtrl.isModalLoading()">
+            <cui-clipboard data-text="ip" data-ng-repeat="ip in IpLoadBalancerFailoverIpDetailCtrl.ips.data track by $index"></cui-clipboard>
+        </cui-clipboard-list>
+
+    </cui-modal-body>
+    <cui-modal-footer>
+        <cui-button data-text="'common_modal_ok' | translate"
```
Comment (whispyy) [inline client/app/iplb/modal/failover-ip/iplb-failover-ip-detail.html]: cui-submit-button

File: client/app/iplb/modal/nat-ip/iplb-nat-ip-detail.html [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+            <cui-modal-text data-text="IpLoadBalancerNatIpDetailCtrl.ips"></cui-modal-text>
+        </div>
+    </cui-modal-body>
+    <cui-modal-footer>
+        <cui-button data-text="'common_modal_ok' | translate"
```
Comment (whispyy) [inline client/app/iplb/modal/nat-ip/iplb-nat-ip-detail.html]: cui-submit-button

File: client/app/iplb/server/preview/iplb-server-preview.html [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+            <dt class="preview-list__term"
+                data-translate="iplb_server_preview_id"></dt>
+            <dd class="preview-list__definition"
+                data-ng-bind=":: IpLoadBalancerServerPreviewCtrl.server.serverId"></dd>
+        </dl>
+    </cui-modal-body>
+    <cui-modal-footer>
+        <cui-button data-text="'common_modal_ok' | translate"
+            data-on-click="IpLoadBalancerServerPreviewCtrl.dismiss()"
+            type="primary"></cui-button>
```
Comment (whispyy) [inline client/app/iplb/server/preview/iplb-server-preview.html]: data-type

File: client/app/iplb/server/status/iplb-server-status-detail.html [hunk 21]
```diff
@@ -0,0 +1,40 @@
...
+            <dt class="preview-list__term"
+                data-translate="iplb_server_status_check_time"></dt>
+            <dd class="preview-list__definition"
+                data-ng-bind=":: serverState.checkTime | date:'short' || '-'"></dd>
+        </dl>
+    </cui-modal-body>
+    <cui-modal-footer>
+        <cui-button data-text="'common_modal_ok' | translate"
+            data-on-click="IpLoadBalancerServerStatusDetailCtrl.dismiss()"
+            type="primary"></cui-button>
```
Comment (whispyy) [inline client/app/iplb/server/status/iplb-server-status-detail.html]: data-type

File: client/app/iplb/serverFarm/preview/iplb-server-farm-preview.html [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+            <dt class="preview-list__term"
+                data-translate="iplb_farm_preview_id"></dt>
+            <dd class="preview-list__definition"
+                data-ng-bind=":: IpLoadBalancerServerFarmPreviewCtrl.farm.farmId"></dd>
+        </dl>
+    </cui-modal-body>
+    <cui-modal-footer>
+        <cui-button data-text="'common_modal_ok' | translate"
+            data-on-click="IpLoadBalancerServerFarmPreviewCtrl.dismiss()"
+            type="primary"></cui-button>
```
Comment (whispyy) [inline client/app/iplb/serverFarm/preview/iplb-server-farm-preview.html]: data-type

File: client/app/iplb/serverFarm/probe/iplb-server-farm-probe.html [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+                class="oui-input"
+                data-ng-model="IpLoadBalancerServerFarmProbeEditCtrl.farmProbe.interval" />
+            <label for="probe_interval" class="oui-label" data-translate="iplb_farm_edit_probe_interval"></label>
+        </div>
+    </cui-modal-body>
+    <cui-modal-footer>
+        <cui-cancel-button data-text="'common_cancel' | translate" data-on-click="IpLoadBalancerServerFarmProbeEditCtrl.dismiss()"></cui-cancel-button>
+        <cui-button data-text="'common_modal_ok' | translate"
+            data-on-click="IpLoadBalancerServerFarmProbeEditCtrl.close()"
+            type="primary"></cui-button>
```
Comment (whispyy) [inline client/app/iplb/serverFarm/probe/iplb-server-farm-probe.html]: data-type

File: client/app/iplb/frontends/preview/iplb-frontends-preview.html [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+            <dt class="preview-list__term"
+                data-translate="iplb_frontend_preview_id"></dt>
+            <dd class="preview-list__definition"
+                data-ng-bind="IpLoadBalancerFrontendPreviewCtrl.frontend.frontendId"></dd>
+        </dl>
+    </cui-modal-body>
+    <cui-modal-footer>
+        <cui-button data-text="'common_modal_ok' | translate"
+            data-on-click="IpLoadBalancerFrontendPreviewCtrl.dismiss()"
+            type="primary"></cui-button>
```
Comment (whispyy) [inline client/app/iplb/frontends/preview/iplb-frontends-preview.html]: data-type

File: client/components/sidebar/translations/Messages_fr_FR.xml [hunk 25]
```diff
@@ -7,6 +7,7 @@
    <translation id="cloud_sidebar_section_license" qtlid="113658">Licences</translation>
    <translation id="cloud_sidebar_section_ip" qtlid="25615">IP</translation>
    <translation id="cloud_sidebar_section_vrack" qtlid="246962">vRack</translation>
+   <translation id="cloud_sidebar_section_load_balancer">IP Loader Balancer</translation>
```
Comment (kumako) [inline client/components/sidebar/translations/Messages_fr_FR.xml]: Load

File: client/app/common/cloud-message.service.js [hunk 26]
```diff
@@ -0,0 +1,83 @@
+class CloudMessage {
```
Comment (neolitec) [inline client/app/common/cloud-message.service.js]: Yes, we sync-ed our branches to use the same components.

File: bower.json [hunk 27]
```diff
@@ -9,6 +9,7 @@
     "jquery": "2.1.x",
     "angular": "1.6.x",
     "ovh-ui-angular": "^1.0.2",
+    "list-view": "ssh://git@stash.ovh.net:7999/uxcomponents/list-view.git#oui-table-2",
```
Comment (neolitec) [inline bower.json]: Yes opensource is done, so we're able to use a public URL.

File: client/app/iplb/iplb-metrics.service.js (hunk 4 above)
Comment (neolitec) [inline client/app/iplb/iplb-metrics.service.js]: YES :D

File: client/app/iplb/modal/offer/iplb-offer-change.html (hunk 8 above)
Comment (neolitec) [inline client/app/iplb/modal/offer/iplb-offer-change.html]: Work in progress...

File: client/app/helper/service-helper.service.js [hunk 28]
```diff
@@ -0,0 +1,46 @@
...
+        constructor ($q, $translate, CloudMessage) {
+            this.$q = $q;
+            this.$translate = $translate;
+            this.CloudMessage = CloudMessage;
+        }
+
+
+        errorHandler (message) {
+            return err => {
+                console.error(err);
```
Comment (neolitec) [inline client/app/helper/service-helper.service.js]: ok

File: client/app/iplb/configuration/iplb-configuration.html [hunk 29]
```diff
@@ -0,0 +1,24 @@
...
+        rows="ctrl.zones.data"
+        on-selection-change="ctrl.onSelectionChange($selection)">
+        <column title="'iplb_configuration_datacenter' | translate" property="name" sortable></column>
+        <column title="'iplb_configuration_progress' | translate"></column>
+        <column title="'iplb_configuration_status' | translate"></column>
+        <column title="'iplb_configuration_action' | translate"
+            class="oui-table__cell_action oui-table__cell_action-menu"
+            template="ctrl.actionTemplate()">
+        </column>
+        <empty-placeholder>Pas de configuration à appliquer / Tout est à jour</empty-placeholder>
```
Comment (neolitec) [inline client/app/iplb/configuration/iplb-configuration.html]: Work in progress...

File: client/app/iplb/frontends/iplb-frontends-edit.controller.js [hunk 30]
```diff
@@ -0,0 +1,244 @@
...
+            case "tls":
+                return "tcp";
+            default:
+                return this.protocol;
+        }
+    }
+
+    getFarmName (farm) {
+        const farmName = farm.displayName || farm.farmId;
+        const farmType = this.$translate.instant(`iplb_frontend_add_protocol_${farm.type}`);
```
Comment (neolitec) [inline client/app/iplb/frontends/iplb-frontends-edit.controller.js]: ok

File: client/app/iplb/frontends/iplb-frontends-edit.controller.js [hunk 31]
```diff
@@ -0,0 +1,244 @@
...
+        if (farm.farmId > 0) {
+            return `${farmName} (${farmType})`;
+        }
+        return farmName;
+    }
+
+    getCertificateName (certificate) {
+        if (certificate.id <= 0 || certificate.id === null) {
```
Comment (neolitec) [inline client/app/iplb/frontends/iplb-frontends-edit.controller.js]: ok

File: client/app/iplb/graph/iplb-graph.html [hunk 32]
```diff
@@ -0,0 +1,39 @@
...
+    <cui-radio-bar data-ng-model="ctrl.scale"
+        data-on-change="ctrl.onScaleChange()">
+        <cui-radio-bar-item data-ng-repeat="(scale, value) in ctrl.scales"
+            data-text="('iplb_graph_scape_' + scale) | translate"
+            data-value="scale"></cui-radio-bar-item>
+    </cui-radio-bar>
+
+    <div class="cui-panel">
+        <div class="cui-panel__header">
+            <h2>Connexions</h2>
```
Comment (neolitec) [inline client/app/iplb/graph/iplb-graph.html]: ok

File: client/app/iplb/graph/iplb-graph.html [hunk 33]
```diff
@@ -0,0 +1,39 @@
...
+                data-chart-data="ctrl.connLoader.data.data"
+                data-chart-labels="ctrl.connLoader.data.labels"
+                data-chart-options="ctrl.options">
+            </canvas>
+        </div>
+    </div>
+
+    <div class="cui-panel">
+        <div class="cui-panel__header">
+            <h2>Requêtes par seconde</h2>
```
Comment (neolitec) [inline client/app/iplb/graph/iplb-graph.html]: ok

File: client/app/iplb/iplb-nat-ip.service.js (hunk 5 above)
Comment (neolitec) [inline client/app/iplb/iplb-nat-ip.service.js]: not any

File: client/app/iplb/home/iplb-home.service.js (hunk 2 above)
Comment (neolitec) [inline client/app/iplb/home/iplb-home.service.js]: ok

File: client/app/iplb/home/iplb-home.html [hunk 34]
```diff
@@ -0,0 +1,119 @@
...
+                    data-actions="ctrl.IPLBActions"></iplb-status-card>
+                <iplb-status-card
+                    data-type="ctrl.frontendsStatus.data.status"
+                    data-title="'Frontends'"
+                    data-subtitle="ctrl.frontendsStatus.data.statusText"
+                    data-loading="ctrl.frontendsStatus.loading"
+                    data-actions="ctrl.frontendsActions"></iplb-status-card>
+                <iplb-status-card
+                    data-type="ctrl.serverFarmsStatus.data.serverFarms.status"
+                    data-title="'Fermes de Serveurs'"
```
Comment (neolitec) [inline client/app/iplb/home/iplb-home.html]: ok

File: client/app/iplb/modal/nat-ip/iplb-nat-ip-detail.html (hunk 6 above)
Comment (neolitec) [inline client/app/iplb/modal/nat-ip/iplb-nat-ip-detail.html]: Work in progress...

File: client/app/iplb/modal/offer/iplb-offer-change.controller.js (hunk 7 above)
Comment (neolitec) [inline client/app/iplb/modal/offer/iplb-offer-change.controller.js]: Work in progress...

File: client/app/iplb/configuration/iplb-configuration.controller.js (hunk 15 above)
Comment (neolitec) [inline client/app/iplb/configuration/iplb-configuration.controller.js]: >$onInit() - Called on each controller after all the controllers on an element have been constructed and had their bindings initialized (and before the pre & post linking functions for the directives on this element). This is a good place to put initialization code for your controller.

Comment (neolitec) [inline client/app/iplb/configuration/iplb-configuration.controller.js]: API calls are actually made in $onInit.

File: client/app/iplb/frontends/iplb-frontends-edit.controller.js [hunk 35]
```diff
@@ -0,0 +1,244 @@
...
+            zone: "0",
+            defaultSslId: 0,
+            defaultFarmId: 0,
+            port: 80,
+            ssl: false
+        };
+        this.type = "http";
+        this.protocol = "http";
+        this.saving = false;
+        this.protocols = this.IpLoadBalancerConstant.protocols;
```
Comment (neolitec) [inline client/app/iplb/frontends/iplb-frontends-edit.controller.js]: >$onInit() - Called on each controller after all the controllers on an element have been constructed and had their bindings initialized (and before the pre & post linking functions for the directives on this element). This is a good place to put initialization code for your controller.

File: client/app/iplb/frontends/iplb-frontends-edit.controller.js (hunk 3 above)
Comment (neolitec) [inline client/app/iplb/frontends/iplb-frontends-edit.controller.js]: ok

File: client/app/iplb/frontends/iplb-frontends-edit.controller.js (hunk 9 above)
Comment (neolitec) [inline client/app/iplb/frontends/iplb-frontends-edit.controller.js]: ok

File: client/app/iplb/home/iplb-home.controller.js (hunk 10 above)
Comment (neolitec) [inline client/app/iplb/home/iplb-home.controller.js]: ok

File: client/app/iplb/server/iplb-server-edit.controller.js (hunk 11 above)
Comment (neolitec) [inline client/app/iplb/server/iplb-server-edit.controller.js]: ok

File: client/app/iplb/server/iplb-server-edit.controller.js (hunk 12 above)
Comment (neolitec) [inline client/app/iplb/server/iplb-server-edit.controller.js]: ok

File: client/app/iplb/serverFarm/iplb-server-farm-edit.controller.js (hunk 13 above)
Comment (neolitec) [inline client/app/iplb/serverFarm/iplb-server-farm-edit.controller.js]: Same comment.

File: client/app/iplb/frontends/iplb-frontends-edit.html (hunk 16 above)
Comment (neolitec) [inline client/app/iplb/frontends/iplb-frontends-edit.html]: ok

File: client/app/iplb/modal/failover-ip/iplb-failover-ip-detail.html (hunk 18 above)
Comment (neolitec) [inline client/app/iplb/modal/failover-ip/iplb-failover-ip-detail.html]: ok

File: client/app/iplb/modal/nat-ip/iplb-nat-ip-detail.html (hunk 19 above)
Comment (neolitec) [inline client/app/iplb/modal/nat-ip/iplb-nat-ip-detail.html]: ok

File: client/app/iplb/server/preview/iplb-server-preview.html (hunk 20 above)
Comment (neolitec) [inline client/app/iplb/server/preview/iplb-server-preview.html]: ok

File: client/app/iplb/serverFarm/preview/iplb-server-farm-preview.html (hunk 22 above)
Comment (neolitec) [inline client/app/iplb/serverFarm/preview/iplb-server-farm-preview.html]: ok

File: client/app/iplb/frontends/preview/iplb-frontends-preview.html (hunk 24 above)
Comment (neolitec) [inline client/app/iplb/frontends/preview/iplb-frontends-preview.html]: ok

File: client/components/sidebar/translations/Messages_fr_FR.xml (hunk 25 above)
Comment (neolitec) [inline client/components/sidebar/translations/Messages_fr_FR.xml]: ok

**Pull request merged**